tests/beaker_tests/cisco_bgp/test_bgp.rb

```diff
@@ -0,0 +1,267 @@
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    # These properties are only configurable in default vrf
+    disable_policy_batching: 'true',
+    event_history_cli:       'size_medium',
+    event_history_detail:    'size_large',
+    event_history_events:    'size_disable',
+    event_history_periodic:  'false',
+  },
+}
+
+tests[:non_def_2] = {
+  desc:           '2.2 Non Defaults Part 2',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    bestpath_always_compare_med:     'true',
+    bestpath_aspath_multipath_relax: 'true',
+    bestpath_compare_routerid:       'true',
+    bestpath_cost_community_ignore:  'true',
+    bestpath_med_confed:             'true',
+    bestpath_med_missing_as_worst:   'true',
+    bestpath_med_non_deterministic:  'true',
+  },
+}
+
+tests[:non_def_3] = {
+  desc:           '2.3 Non Defaults Part 3',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    cluster_id:                             '10.0.0.1',
+    confederation_id:                       '99',
+    confederation_peers:                    '55 23.4 88 200.1',
+    enforce_first_as:                       'true',
+    fast_external_fallover:                 'true',
+    flush_routes:                           'false',
+    graceful_restart:                       'true',
+    graceful_restart_helper:                'true',
+    graceful_restart_timers_restart:        '130',
+    graceful_restart_timers_stalepath_time: '310',
+  },
+}
+
+tests[:non_def_4] = {
+  desc:           '2.4 Non Defaults Part 4',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    isolate:                     'false',
+    log_neighbor_changes:        'true',
+    maxas_limit:                 '50',
+    router_id:                   '192.168.0.66',
+    shutdown:                    'true',
+    suppress_fib_pending:        'true',
+    timer_bestpath_limit:        '255',
+    timer_bestpath_limit_always: 'true',
+    timer_bgp_holdtime:          '110',
+    timer_bgp_keepalive:         '45',
+  },
+}
+
+# Platform-specific tests
+tests[:non_def_plat_1] = {
+  desc:           '3.1 Default Properties Platform-specific Part 1',
+  platform:       'n(3|9)k',
+  title_pattern:  "#{asn} default",
+  manifest_props: {
+    neighbor_down_fib_accelerate: 'true'
+  },
+}
+
+tests[:title_patterns] = {
+  preclean:       'cisco_bgp',
+  manifest_props: {},
+  resource:       { 'ensure' => 'present' },
+}
+
+tests[:title_patterns_1] = {
+  desc:          'T.1 Title Pattern',
+  preclean:      'cisco_bgp',
+  title_pattern: 'new_york',
+  title_params:  { asn: '11.4', vrf: 'red' },
+  resource:      { 'ensure' => 'present' },
+}
+
+tests[:title_patterns_2] = {
+  desc:          'T.2 Title Pattern',
+  title_pattern: '11.4',
+  title_params:  { vrf: 'blue' },
+  resource:      { 'ensure' => 'present' },
+}
+
+# This helper tests a test case in vrf context. This allows for testing a vrf
+# while an existing config is present in vrf default.
```

Comment:
I included this because the old tests had coverage for it but I'm not convinced that these extra vrf tests are really needed. This just adds a different context to the previous tests. Plus we've already got vrf testing in minitest. I'd be happy to remove them.